docs_src/examples/Ch-OSImageWithEdgeX.md

```diff
@@ -0,0 +1,867 @@
+    },
+    {
+      "id": "2c1ea95a-b5b0-4051-a5b6-24d98f40a5c3",
+      "origin": 1660314776365793500,
+      "deviceName": "Random-Integer-Device",
+      "resourceName": "Int64",
+      "profileName": "Random-Integer-Device",
+      "valueType": "Int64",
+      "value": "-8755202057804760537"
+    }
+  ]
+}
+```
+
+The response is similar to if query was sent directly to Core Data (http://localhost:59880/api/v2/reading/all?limit=2), except that it was done over TLS and with authentication. We didn't need to bind the Core Data's server to other interfaces or expose its port to outside.
+
+!!! tip
+    EdgeX's API Gateway internally generated a self-signed TLS certificate. The system doesn't trust that certificate and that's why we set the `--insecure` flag for curl.
+
+    Refer [here](../../getting-started/Ch-GettingStartedSnapUsers/#changing-tls-certificates) to learn how you can replace the certificate with one that your system trusts.
+
+In this chapter, we created an OS image which comes with EdgeX components that are automatically started and configured to securely receive external requests. 
+We can extend the server configurations by setting other defaults in the gadget.
+This is sufficient in most scenarios and allows creating an image that is configured and ready for production.
+
+The server configuration is made possible via a combination of snap options and environment variable overrides implemented for EdgeX services. There are two situations in which we need to override entire configuration files instead of fields one by one:
+
+1. When we have to easily override many configuration fields: it is very cumbersome to override many configuration fields one by one.
+2. When we need to add or change device, profile, and provision watcher configurations.
+
+For the above cases, we need to supply whole configuration files to applications.
+In the next chapter, we walk through creating a Snap package with our custom configuration files. The package will become part of the OS image and supply necessary configurations to all other EdgeX applications.
+
+## C. Override configuration files
+
+This chapter builds on top of what we did previously and shows how to override entire configuration files with a packaged copy, prepared for an specific use case.
+
+### Create a config provider for Device Virtual
+The EdgeX Device Virtual service cannot be fully configured using environment variables / snap options. Because of that, we need to package the modified config files and replace the defaults.
+Moreover, it is tedious to override many configurations one by one, compared to having a file which contains all the needed modifications.
+
+Since we want to create an OS image pre-loaded with the configured system, we need to make sure the configurations are there without any manual user interaction. We do that by creating a snap which provides the configuration files to the Device Virtual snap:
+
+- configuration.toml
+- devices
+- profiles
```

Comment:
Aren't these files also `.toml` files?

Reply:
Those are directories. Will update the text to make that clear.